File: java_projects/glue_code/commons-validator/src/main/java/org/apache/commons/validator/GenericValidator.java
package org.apache.commons.validator;

import java.io.Serializable;
import java.util.Locale;
import org.apache.commons.validator.routines.CreditCardValidator;
import org.apache.commons.validator.routines.UrlValidator;
import org.graalvm.polyglot.Value;

public class GenericValidator implements Serializable {
  private static final CreditCardValidator CREDIT_CARD_VALIDATOR =
      CreditCardValidator.CreditCardValidator0();
  private static final UrlValidator URL_VALIDATOR = UrlValidator.UrlValidator6();
  private static final long serialVersionUID = -7212095066891517618L;
  private static Value clz =
      ContextInitializer.getPythonClass("/GenericValidator.py", "GenericValidator");
  private Value obj;

  public GenericValidator(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public static boolean maxValue3(float value, float max) {
    //
    // return (value <= max);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("maxValue3", value, max).as(boolean.class);
  }

  public static boolean maxValue2(double value, double max) {
    //
    // return (value <= max);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("maxValue2", value, max).as(boolean.class);
  }

  public static boolean maxValue1(long value, long max) {
    //
    // return (value <= max);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("maxValue1", value, max).as(boolean.class);
  }

  public static boolean maxValue0(int value, int max) {
    //
    // return (value <= max);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("maxValue0", value, max).as(boolean.class);
  }

  public static boolean minValue3(float value, float min) {
    //
    // return (value >= min);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("minValue3", value, min).as(boolean.class);
  }

  public static boolean minValue2(double value, double min) {
    //
    // return (value >= min);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("minValue2", value, min).as(boolean.class);
  }

  public static boolean minValue1(long value, long min) {
    //
    // return (value >= min);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("minValue1", value, min).as(boolean.class);
  }

  public static boolean minValue0(int value, int min) {
    //
    // return (value >= min);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("minValue0", value, min).as(boolean.class);
  }

  public static boolean minLength1(String value, int min, int lineEndLength) {
    //
    // int adjustAmount = adjustForLineEnding(value, lineEndLength);
    // return ((value.length() + adjustAmount) >= min);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("minLength1", value, min, lineEndLength).as(boolean.class);
  }

  public static boolean minLength0(String value, int min) {
    //
    // return (value.length() >= min);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("minLength0", value, min).as(boolean.class);
  }

  public static boolean maxLength1(String value, int max, int lineEndLength) {
    //
    // int adjustAmount = adjustForLineEnding(value, lineEndLength);
    // return ((value.length() + adjustAmount) <= max);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("maxLength1", value, max, lineEndLength).as(boolean.class);
  }

  public static boolean maxLength0(String value, int max) {
    //
    // return (value.length() <= max);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("maxLength0", value, max).as(boolean.class);
  }

  public static boolean isUrl(String value) {
    //
    // return URL_VALIDATOR.isValid(value);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isUrl", value).as(boolean.class);
  }

  public static boolean isEmail(String value) {
    //
    // return EmailValidator.getInstance0().isValid(value);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isEmail", value).as(boolean.class);
  }

  public static boolean isCreditCard(String value) {
    //
    // return CREDIT_CARD_VALIDATOR.isValid(value);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isCreditCard", value).as(boolean.class);
  }

  public static boolean isInRange5(double value, double min, double max) {
    //
    // return ((value >= min) && (value <= max));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInRange5", value, min, max).as(boolean.class);
  }

  public static boolean isInRange4(long value, long min, long max) {
    //
    // return ((value >= min) && (value <= max));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInRange4", value, min, max).as(boolean.class);
  }

  public static boolean isInRange3(short value, short min, short max) {
    //
    // return ((value >= min) && (value <= max));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInRange3", value, min, max).as(boolean.class);
  }

  public static boolean isInRange2(float value, float min, float max) {
    //
    // return ((value >= min) && (value <= max));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInRange2", value, min, max).as(boolean.class);
  }

  public static boolean isInRange1(int value, int min, int max) {
    //
    // return ((value >= min) && (value <= max));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInRange1", value, min, max).as(boolean.class);
  }

  public static boolean isInRange0(byte value, byte min, byte max) {
    //
    // return ((value >= min) && (value <= max));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInRange0", value, min, max).as(boolean.class);
  }

  public static boolean isDate1(String value, String datePattern, boolean strict) {
    //
    // return org.apache.commons.validator.DateValidator.getInstance()
    // .isValid0(value, datePattern, strict);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isDate1", value, datePattern, strict).as(boolean.class);
  }

  public static boolean isDate0(String value, Locale locale) {
    //
    // return DateValidator.getInstance().isValid2(value, locale);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isDate0", value, locale).as(boolean.class);
  }

  public static boolean isDouble(String value) {
    //
    // return (GenericTypeValidator.formatDouble0(value) != null);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isDouble", value).as(boolean.class);
  }

  public static boolean isFloat(String value) {
    //
    // return (GenericTypeValidator.formatFloat0(value) != null);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isFloat", value).as(boolean.class);
  }

  public static boolean isLong(String value) {
    //
    // return (GenericTypeValidator.formatLong0(value) != null);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isLong", value).as(boolean.class);
  }

  public static boolean isInt(String value) {
    //
    // return (GenericTypeValidator.formatInt0(value) != null);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isInt", value).as(boolean.class);
  }

  public static boolean isShort(String value) {
    //
    // return (GenericTypeValidator.formatShort0(value) != null);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isShort", value).as(boolean.class);
  }

  public static boolean isByte(String value) {
    //
    // return (GenericTypeValidator.formatByte0(value) != null);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isByte", value).as(boolean.class);
  }

  public static boolean matchRegexp(String value, String regexp) {
    //
    // if (regexp == null || regexp.length() <= 0) {
    // return false;
    // }
    //
    // return Pattern.matches(regexp, value);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("matchRegexp", value, regexp).as(boolean.class);
  }

  public static boolean isBlankOrNull(String value) {
    //
    // return ((value == null) || (value.trim().length() == 0));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("isBlankOrNull", value).as(boolean.class);
  }

  private static int adjustForLineEnding(String value, int lineEndLength) {
    //
    // int nCount = 0;
    // int rCount = 0;
    // for (int i = 0; i < value.length(); i++) {
    // if (value.charAt(i) == '\n') {
    // nCount++;
    // }
    // if (value.charAt(i) == '\r') {
    // rCount++;
    // }
    // }
    // return ((nCount * lineEndLength) - (rCount + nCount));
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("adjustForLineEnding", value, lineEndLength).as(int.class);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/body/commons-validator/src/main/org/apache/commons/validator/GenericValidator.py
from __future__ import annotations
import locale
import pathlib
import io
import typing
from typing import *
from src.main.org.apache.commons.validator.DateValidator import *
from src.main.org.apache.commons.validator.GenericTypeValidator import *
from src.main.org.apache.commons.validator.routines.CreditCardValidator import *
from src.main.org.apache.commons.validator.routines.DateValidator import *
from src.main.org.apache.commons.validator.routines.EmailValidator import *
from src.main.org.apache.commons.validator.routines.UrlValidator import *


class GenericValidator:

    __CREDIT_CARD_VALIDATOR: CreditCardValidator = (
        CreditCardValidator.CreditCardValidator0()
    )
    __URL_VALIDATOR: UrlValidator = UrlValidator.UrlValidator6()
    __serialVersionUID: int = -7212095066891517618

    @staticmethod
    def maxValue3(value: float, max: float) -> bool:
        return value <= max

    @staticmethod
    def maxValue2(value: float, max: float) -> bool:
        return value <= max

    @staticmethod
    def maxValue1(value: int, max: int) -> bool:
        return value <= max

    @staticmethod
    def maxValue0(value: int, max: int) -> bool:
        return value <= max

    @staticmethod
    def minValue3(value: float, min: float) -> bool:
        return value >= min

    @staticmethod
    def minValue2(value: float, min: float) -> bool:
        return value >= min

    @staticmethod
    def minValue1(value: int, min: int) -> bool:
        return value >= min

    @staticmethod
    def minValue0(value: int, min: int) -> bool:
        return value >= min

    @staticmethod
    def minLength1(value: str, min: int, lineEndLength: int) -> bool:

        adjustAmount: int = GenericValidator.__adjustForLineEnding(value, lineEndLength)
        return (len(value) + adjustAmount) >= min

    @staticmethod
    def minLength0(value: str, min: int) -> bool:
        return len(value) >= min

    @staticmethod
    def maxLength1(value: str, max: int, lineEndLength: int) -> bool:

        adjustAmount: int = GenericValidator.__adjustForLineEnding(value, lineEndLength)
        return (len(value) + adjustAmount) <= max

    @staticmethod
    def maxLength0(value: str, max: int) -> bool:
        return len(value) <= max

    @staticmethod
    def isUrl(value: str) -> bool:

        if value is None:
            return False

        try:
            uri = urlparse(value)
        except ValueError:
            return False

        scheme = uri.scheme
        if not GenericValidator.__URL_VALIDATOR._isValidScheme(scheme):
            return False

        authority = uri.netloc
        if scheme.lower() == "file" and (authority is None or authority == ""):
            return True
        elif scheme.lower() == "file" and authority is not None and ":" in authority:
            return False
        else:
            if not GenericValidator.__URL_VALIDATOR._isValidAuthority(authority):
                return False

        if not GenericValidator.__URL_VALIDATOR._isValidPath(uri.path):
            return False

        if not GenericValidator.__URL_VALIDATOR._isValidQuery(uri.query):
            return False

        if not GenericValidator.__URL_VALIDATOR._isValidFragment(uri.fragment):
            return False

        return True

    @staticmethod
    def isEmail(value: str) -> bool:
        return EmailValidator.getInstance0().isValid(value)

    @staticmethod
    def isCreditCard(value: str) -> bool:
        return GenericValidator.__CREDIT_CARD_VALIDATOR.isValid(value)

    @staticmethod
    def isInRange5(value: float, min: float, max: float) -> bool:
        return (value >= min) and (value <= max)

    @staticmethod
    def isInRange4(value: int, min: int, max: int) -> bool:
        return (value >= min) and (value <= max)

    @staticmethod
    def isInRange3(value: int, min: int, max: int) -> bool:
        return (value >= min) and (value <= max)

    @staticmethod
    def isInRange2(value: float, min: float, max: float) -> bool:
        return (value >= min) and (value <= max)

    @staticmethod
    def isInRange1(value: int, min: int, max: int) -> bool:
        return (value >= min) and (value <= max)

    @staticmethod
    def isInRange0(value: int, min: int, max: int) -> bool:
        return (value >= min) and (value <= max)

    @staticmethod
    def isDate1(value: str, datePattern: str, strict: bool) -> bool:
        return DateValidator.getInstance().isValid0(value, datePattern, strict)

    @staticmethod
    def isDate0(value: str, locale: typing.Any) -> bool:
        return DateValidator.getInstance().isValid2(value, locale)

    @staticmethod
    def isDouble(value: str) -> bool:
        return GenericTypeValidator.formatDouble0(value) is not None

    @staticmethod
    def isFloat(value: str) -> bool:
        return GenericTypeValidator.formatFloat0(value) is not None

    @staticmethod
    def isLong(value: str) -> bool:

        return GenericTypeValidator.formatLong0(value) is not None

    @staticmethod
    def isInt(value: str) -> bool:
        return GenericTypeValidator.formatInt0(value) is not None

    @staticmethod
    def isShort(value: str) -> bool:
        return GenericTypeValidator.formatShort0(value) is not None

    @staticmethod
    def isByte(value: str) -> bool:

        return GenericTypeValidator.formatByte0(value) is not None

    @staticmethod
    def matchRegexp(value: str, regexp: str) -> bool:

        import re

        if regexp is None or len(regexp) <= 0:
            return False

        return bool(re.match(regexp, value))

    @staticmethod
    def isBlankOrNull(value: str) -> bool:
        return (value is None) or (len(value.strip()) == 0)

    @staticmethod
    def __adjustForLineEnding(value: str, lineEndLength: int) -> int:

        nCount: int = 0
        rCount: int = 0

        for i in range(len(value)):
            if value[i] == "\n":
                nCount += 1
            if value[i] == "\r":
                rCount += 1

        return (nCount * lineEndLength) - (rCount + nCount)
